uxarray/core/dataarray.py

```diff
@@ -282,3 +282,11 @@ def integrate(self,
                            name=self.name)
 
         return uxda
+
+    def face_centered(self) -> bool:
+        return (self.uxgrid.nMesh2_face == self.shape[-1] and
+                self.uxgrid.nMesh2_node not in self.shape)
+
+    def node_centered(self) -> bool:
+        return (self.uxgrid.nMesh2_node == self.shape[-1] and
+                self.uxgrid.nMesh2_face not in self.shape)
```

Comment:
Let us keep these two functions in the internal API (and not forget to list them under docs/internal/index.rst). 

This is because there might not be any need to have such user API functions since the task they perform is straightforward for the use by just checking the array dimensions and comparing them to grid values (but calculating that info becomes vital for the visualization functions like here)

Reply:
> This is because there might not be any need to have such user API functions since the task they perform is straightforward for the use by just checking the array dimensions and comparing them to grid values (but calculating that info becomes vital for the visualization functions like here)

While it may not seem like there is a need, I also don't see any harm in exposing it to the user. However, if we still want to keep it internal I can make that change.

Reply:
One concern is that the user API doc gets overpopulated and less readable/helpful. So, I'd still ask to keep them internal unless there is explicit request from the user. Would that make sense?

Reply:
That does make sense, but typically the API documentation isn't mean to be "readable", it should be a comprehensive list that someone can quickly search for. 

The readability comes from our usage example and (soon) the user guides.

Reply:
I agree with the comprehensiveness of the API doc but wouldn't agree with that it's not meant to be readable. One should be able to practically browse user API docs and sometimes read a number of functions (even sometimes in in a sequence just to explore) and shouldn't feel overwhelmed due to too many things. That said, my main point here is that we shouldn't overpopulate our user API with too many functions regardless of if they'd be used or not by the end-user. 

I believe these functions that check if data is face- or node-centered are good internal API functions but not needed in the user API.

Reply:
I see your point now! This makes sense.

I'll add a docstring to them and make them internal